Prepare meilisearch 2021-w1 (#673)
meilisearch/documentation: guides/advanced_guides/tokenization.md

> # Tokenization

Reviewer: ```suggestion
# Tokenization and Language Handling
```
Maybe a way to make it easier to understand for users that do not know the word tokenization? This is really just a suggestion. But I feel like you really have to know about the concept to know this word. In which case you wouldn't have to go in this guide cause you already know the concept.
And for users that do not know about tokenization, I suppose they will not have a reason to go on this guide. Except if it is linked somewhere that creates a context.

Author: 100% yes :) It's such a nice feeling when someone suggests a change you were already making remotely :)

Reviewer: Please, feel free to commit :)

**Tokenization** is the act of taking a sentence or phrase and splitting it into smaller units of language, called tokens. It is the first step of document indexing in the Meilisearch engine, and is a critical factor in the quality of search results.

Breaking sentences into smaller chunks requires understanding where one word ends and another begins, making tokenization a highly complex and language-dependant task. MeiliSearch's solution to this problem is a **modular tokenizer** that follows different processes, called **pipelines**, based on the type of character it detects. This allows it to function on datasets from multiple languages with minimal setup.

> Currently, the MeiliSearch tokenizer has full support for four languages: English, Chinese, Japanese, and Korean. It can also effectively tokenize any language that uses the Latin alphabet.

Reviewer: ```suggestion
Currently, the MeiliSearch tokenizer has full support for four languages: 
- English  🇬🇧 (and all Latin based languages)
- Chinese 🇨🇳
- Japanese 🇯🇵
- Korean 🇰🇷
```
Maybe an emphasis on the supported language? 
This should be added as well in the readme.md of the core meilisearch repo: https://github.com/meilisearch/meilisearch (do you want me to make an issue about it?)

Author: Great change. Yes, please :)

Reviewer: Since you are the creator of the PR, you should make the commits. If you like my suggestion you can apply it by clicking here: 
<img width="861" alt="Screenshot 2021-01-11 at 21 30 07" src="https://user-images.githubusercontent.com/33010418/104234770-3817c500-5454-11eb-9463-5fb6d21962c2.png">

> ## The Meili Tokenizer

Reviewer: ```suggestion
## Deep Dive in the MeiliSearch Tokenizer
```
Simple suggestion as this goes deeper than what a common user would want to know.

![Tokenizer Architecture](https://user-images.githubusercontent.com/6482087/102896344-8560d200-4466-11eb-8cfe-b4ae8741093b.jpg)

The tokenization process can be broken down into two broad steps:

> 1. Determine the type of script that must be tokenized (e.g. Latin alphabet, Arabic script, Cyrillic etc.).

Reviewer: ```suggestion
1. Determine the type of script that must be tokenized (e.g. Latin alphabet, Chinese, Korean etc.).
```
Just to not confuse users on what we support
